### PR TITLE
NIFI-11260: Added SSL Context Service in AWSCredentialsProviderContro…

### PR DESCRIPTION
**Pull request by turcsanyip**
…llerService

# Summary

[NIFI-11260](https://issues.apache.org/jira/browse/NIFI-11260)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [x] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/credentials/provider/factory/strategies/AssumeRoleCredentialsStrategy.java [hunk 1]
```diff
@@ -190,29 +202,27 @@ public AWSCredentialsProvider getDerivedCredentialsProvider(final Map<PropertyDe
...
+        final String assumeRoleName = propertyContext.getProperty(ASSUME_ROLE_NAME).getValue();
+        final int maxSessionTime = propertyContext.getProperty(MAX_SESSION_TIME).asInteger();
+        final String assumeRoleExternalId = propertyContext.getProperty(ASSUME_ROLE_EXTERNAL_ID).getValue();
+        final String assumeRoleSTSEndpoint = propertyContext.getProperty(ASSUME_ROLE_STS_ENDPOINT).getValue();
+        final String stsRegion = propertyContext.getProperty(ASSUME_ROLE_STS_REGION).getValue();
 
         final StsAssumeRoleCredentialsProvider.Builder builder = StsAssumeRoleCredentialsProvider.builder();
 
         // If proxy variables are set, then create Client Configuration with those values
         final ApacheHttpClient.Builder httpClientBuilder = ApacheHttpClient.builder();
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/credentials/provider/factory/strategies/AssumeRoleCredentialsStrategy.java:222]: The `org.apache.http.conn.ssl.SSLConnectionFactory` can be configured with an `SSLContext` from `SSLContextService` and provided to the `httpClientBuilder` using the `socketFactory()` method.

**Review by exceptionfactory — COMMENTED** [review]
Thanks for the improvement @turcsanyip. Changing the internal contract from a Map of resolved properties to the PropertyContext makes sense, and supports the option for a custom SSL Context Service.

I noted one suggestion about adding the optional SSL Context service to the AWS V2 `AwsCredentialsProvider` using the `socketFactory()` builder method.

File: nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/credentials/provider/factory/strategies/AssumeRoleCredentialsStrategy.java (hunk 1 above)
Comment (turcsanyip) [reply nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/credentials/provider/factory/strategies/AssumeRoleCredentialsStrategy.java:219]: @exceptionfactory Thanks for the hint! I updated the PR.

**Review by exceptionfactory — APPROVED** [review]
Thanks for making the adjustments @turcsanyip, the latest version looks good! +1 merging

**Pull request closed without merging**